x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/allocation/SearchableSnapshotDiskThresholdIntegTests.java

```diff
@@ -268,10 +283,159 @@ public void testHighWatermarkCanBeExceededOnColdOrFrozenNode() throws Exception
+        logger.info(
+            "--> reducing disk size of node [{}/{}] so that all shards except one can fit on the node",
+            otherDataNode,
+            otherDataNodeId
+        );
+        String indexToSkip = randomFrom(indicesStoresSizes.keySet());
+        Map<String, Long> indicesToBeMounted = indicesStoresSizes.entrySet()
+            .stream()
+            .filter(e -> e.getKey().equals(indexToSkip) == false)
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        long totalSpace = indicesToBeMounted.values().stream().mapToLong(e -> e).sum() + WATERMARK_BYTES + 1024L;
+        getTestFileStore(otherDataNode).setTotalSpace(totalSpace);
+
+        logger.info("--> refreshing cluster info");
+        InternalClusterInfoService masterInfoService = (InternalClusterInfoService) internalCluster().getCurrentMasterNodeInstance(
+            ClusterInfoService.class
+        );
+        ClusterInfoServiceUtils.refresh(masterInfoService);
+        assertThat(
+            masterInfoService.getClusterInfo().getNodeMostAvailableDiskUsages().get(otherDataNodeId).getTotalBytes(),
+            equalTo(totalSpace)
+        );
+
+        String prefix = "mounted-";
+        mountIndices(indicesToBeMounted.keySet(), prefix, repositoryName, snapshotName, FULL_COPY);
+
+        assertBusy(() -> {
+            var state = client().admin().cluster().prepareState().setRoutingTable(true).get().getState();
+            assertThat(
+                state.routingTable()
+                    .allShards()
+                    .stream()
+                    .filter(s -> indicesToBeMounted.containsKey(s.shardId().getIndexName().replace(prefix, "")))
+                    .filter(s -> state.metadata().index(s.shardId().getIndex()).isSearchableSnapshot())
+                    .filter(s -> otherDataNodeId.equals(s.currentNodeId()))
+                    .filter(s -> s.state() == ShardRoutingState.INITIALIZING)
+                    .count(),
+                equalTo((long) indicesToBeMounted.size())
+            );
+        });
+
+        logger.info("--> All shards are being initialized, attempt to mount an extra index");
+
+        mountIndices(List.of(indexToSkip), prefix, repositoryName, snapshotName, FULL_COPY);
+        assertBusy(() -> {
+            var state = client().admin().cluster().prepareState().setRoutingTable(true).get().getState();
+            assertThat(state.routingTable().index(prefix + indexToSkip).shardsWithState(ShardRoutingState.STARTED).size(), equalTo(0));
+        });
+
+        logger.info("--> Unlocking the initialized shards");
+        var mockRepository = (CustomMockRepository) internalCluster().getCurrentMasterNodeInstance(RepositoriesService.class)
+            .repository(repositoryName);
+        mockRepository.unlock();
+
+        assertBusy(() -> {
+            var state = client().admin().cluster().prepareState().setRoutingTable(true).get().getState();
+            assertThat(
```

Comment:
nit: Since there's just one index that should be unassigned, I think it might be clearer to assert that the shard is `UNASSIGNED`?